pkg/roachprod/failureinjection/failures/network_partition.go

```diff
@@ -0,0 +1,239 @@
+) error {
+	if f.c.IsLocal() {
+		l.Printf("Local cluster detected, skipping iptables command")
+		return nil
+	}
+	cmd := strings.Join(args, " ")
+	return f.c.Run(ctx, l, l.Stdout, l.Stderr, install.WithNodes(node), "iptables", cmd)
+}
+
+// runWithDetails is a wrapper function for SyncedCluster.RunWithDetails.
+func (f *IPTablesPartitionNode) runWithDetails(
+	ctx context.Context, l *logger.Logger, node install.Nodes, args ...string,
+) (install.RunResultDetails, error) {
+	if f.c.IsLocal() {
+		l.Printf("Local cluster detected, skipping iptables command")
+		return install.RunResultDetails{}, nil
+	}
+
+	cmd := strings.Join(args, " ")
+	res, err := f.c.RunWithDetails(ctx, l, install.WithNodes(node), "iptables", cmd)
+	if err != nil {
+		return install.RunResultDetails{}, err
+	}
+	return res[0], nil
+}
+
+func (f *IPTablesPartitionNode) Setup(_ context.Context, _ *logger.Logger, _ FailureArgs) error {
+	// iptables is already installed by default on Ubuntu.
+	return nil
+}
+
+// When dropping both input and output, make sure we drop packets in both
+// directions for both the inbound and outbound TCP connections, such that we
+// get a proper black hole. Only dropping one direction for both of INPUT and
+// OUTPUT will still let e.g. TCP retransmits through, which may affect the
+// TCP stack behavior and is not representative of real network outages.
+//
+// For the asymmetric partitions, only drop packets in one direction since
+// this is representative of accidental firewall rules we've seen cause such
+// outages in the wild.
+const (
+	partitionTemplateWrapper = `
+# ensure any failure fails the entire script.
+set -e;
+# Setting default filter policy
+sudo iptables -P INPUT ACCEPT;
+sudo iptables -P OUTPUT ACCEPT;
+%s
+sudo iptables-save
+`
+
+	// Drop all incoming and outgoing traffic to the ip address.
+	bidirectionalPartitionCmd = `
+sudo iptables %[1]s INPUT  -s {ip:%[2]d} -p tcp --dport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s OUTPUT -d {ip:%[2]d} -p tcp --dport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s INPUT  -s {ip:%[2]d} -p tcp --sport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s OUTPUT -d {ip:%[2]d} -p tcp --sport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s INPUT  -s {ip:%[2]d:public} -p tcp --dport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s OUTPUT -d {ip:%[2]d:public} -p tcp --dport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s INPUT  -s {ip:%[2]d:public} -p tcp --sport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s OUTPUT -d {ip:%[2]d:public} -p tcp --sport {pgport:%[2]d} -j DROP;
+`
+	// Drop all incoming traffic from the ip address.
+	asymmetricInputPartitionCmd = `
+sudo iptables %[1]s INPUT -s {ip:%[2]d} -p tcp --dport {pgport:%[2]d} -j DROP;
+sudo iptables %[1]s INPUT -s {ip:%[2]d:public} -p tcp --dport {pgport:%[2]d} -j DROP;
+`
+
+	// Drop all outgoing traffic to the ip address.
+	asymmetricOutputPartitionCmd = `
+sudo iptables %[1]s OUTPUT -s {ip:%[2]d} -p tcp --dport {pgport:%[2]d} -j DROP;
```

Comment:
I think you meant `-d` instead of `-s` on both these rules.

Reply:
Oops! Whole point of the smoke test is to catch silly mistakes like this except I didn't write one for outgoing traffic 🙃. Nice catch! Fixed and added a smoke test.